mmrazor/models/algorithms/nas/bignas.py

```diff
@@ -0,0 +1,342 @@
+            loaded dict or built :obj:`FixSubnet`. Defaults to None.
+        data_preprocessor (Optional[Union[dict, nn.Module]]): The pre-process
+            config of :class:`BaseDataPreprocessor`. Defaults to None.
+        strategy (str): The sampling strategy. Defaults to `sandwich4`.
+        num_samples (int): Nums of the base samples. Defaults to 2.
+        drop_path_rate (float): Stochastic depth rate. Defaults to 0.2.
+        backbone_dropout_stages (List): Stages to be set dropout. Defaults to
+            [6, 7].
+        init_cfg (Optional[dict]): Init config for ``BaseModule``.
+            Defaults to None.
+
+    Note:
+        BigNAS uses two mutators which are ``DynamicValueMutator`` and
+        ``ChannelMutator``. `DynamicValueMutator` handle the mutable object
+        ``OneShotMutableValue`` in BigNAS while ChannelMutator handle
+        the mutable object ``OneShotMutableChannel`` in BigNAS.
+    """
+
+    strategy_groups: Dict[str, List] = {
+        'sandwich2': ['max', 'min'],
+        'sandwich3': ['max', 'random', 'min'],
+        'sandwich4': ['max', 'random', 'random1', 'min'],
+        'batch-sandwich4': [['max', 'random', 'random', 'min']],
+        'batch-sandwich4_distill': [['max'],
+                                    ['max', 'random', 'random', 'min']]
+    }
+
+    def __init__(self,
+                 architecture: Union[BaseModel, Dict],
+                 mutators: VALID_MUTATORS_TYPE,
+                 distiller: VALID_DISTILLER_TYPE,
+                 fix_subnet: Optional[ValidFixMutable] = None,
+                 data_preprocessor: Optional[Union[Dict, nn.Module]] = None,
+                 strategy: str = 'sandwich4',
+                 num_samples: int = 2,
+                 drop_path_rate: float = 0.2,
+                 backbone_dropout_stages: List = [6, 7],
+                 init_cfg: Optional[Dict] = None) -> None:
+        super().__init__(architecture, data_preprocessor, init_cfg)
+
+        if isinstance(mutators, dict):
+            built_mutators: Dict = dict()
+            for name, mutator_cfg in mutators.items():
+                if 'parse_cfg' in mutator_cfg and isinstance(
+                        mutator_cfg['parse_cfg'], dict):
+                    assert mutator_cfg['parse_cfg'][
+                        'type'] == 'Predefined', \
+                            'BigNAS only support predefined.'
+                mutator: BaseMutator = MODELS.build(mutator_cfg)
+                built_mutators[name] = mutator
+                mutator.prepare_from_supernet(self.architecture)
+            self.mutators = built_mutators
+        else:
+            raise TypeError('mutator should be a `dict` but got '
+                            f'{type(mutators)}')
+
+        self.distiller = self._build_distiller(distiller)
+        self.distiller.prepare_from_teacher(self.architecture)
+        self.distiller.prepare_from_student(self.architecture)
+
+        self.strategy = strategy
```

Comment:
if use stategy, it could be more clear that using `num_samples` to denote the total subnets, e.g. self.num_samples = len(self.strategy_groups[strategy]).